scripts/tests/chiptest/test_definition.py

```diff
@@ -159,10 +159,15 @@ def __terminateProcess(self):
         if self.process:
             self.process.terminate()  # sends SIGTERM
             try:
-                self.process.wait(10)
+                exit_code = self.process.wait(10)
+                if exit_code:
+                    raise Exception('Subprocess failed with exit code: %d' % exit_code)
```

Comment:
Why there is an exception raised in case when the `returncode !=0` but not when the process did not respond to `SIGTERM` (which also is some kind of incorrect behavior)?

Reply:
I agree that it is some kind of incorrect behaviour and ideally I would like to raise an exception.
This is not done in this patch for the reason that I would like to stop us from introducing new leaks and right now the leaks report is ignored - which is very bad.
So this particular patch targets this use case and not the one were the process hangs while I agree that we should ultimately raise an exception too.